From b7fdb4125e4eebc5af2155b6d36cf17339a3abf5 Mon Sep 17 00:00:00 2001
From: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
Date: Tue, 25 Dec 2018 14:44:11 +0300
Subject: [PATCH] tests: cipher-openssl-compat: don't call
 EVP_CIPHER_CTX_init()

There is no need to call EVP_CIPHER_CTX_init() after
EVP_CIPHER_CTX_new().

Fixes #658

Signed-off-by: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
---
 tests/slow/cipher-openssl-compat.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tests/slow/cipher-openssl-compat.c b/tests/slow/cipher-openssl-compat.c
index a4cb3522d..64adf25a4 100644
--- a/tests/slow/cipher-openssl-compat.c
+++ b/tests/slow/cipher-openssl-compat.c
@@ -106,7 +106,6 @@ static int cipher_test(const char *ocipher, gnutls_cipher_algorithm_t gcipher,
 #if OPENSSL_VERSION_NUMBER >= 0x10100000L
 		if (gcipher == GNUTLS_CIPHER_AES_128_CCM
 		    || gcipher == GNUTLS_CIPHER_AES_256_CCM) {
-			assert(EVP_CIPHER_CTX_init(ctx)==1);
 			assert(EVP_CipherInit_ex(ctx, evp_cipher, 0, 0, 0, 0) >
 			       0);
 
-- 
2.19.2

